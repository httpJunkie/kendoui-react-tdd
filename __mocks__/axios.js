const axiosMock = jest.genMockFromModule('axios');
let mockResponse = {
  data: {
    "locations": [
      {
        "city": "test city",
        "address": "test address"
      }
    ]
  }
};
axiosMock.get.mockImplementation(req);

function req() {
  return new Promise(function (resolve) {
    axiosMock.delayTimer = setTimeout(function () {
      resolve(mockResponse);
    }, 100);
  });
}

module.exports = axiosMock;
