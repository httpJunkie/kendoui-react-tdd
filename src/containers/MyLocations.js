import React, { Component } from 'react';
import Header from '../components/Header';
import Button from '../components/Button';
import Map from '../components/Map';
import mapChooser from '../mapChooser';
import axios from 'axios';

class MyLocations extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentMap: 'none.png',
      locations: []
    };
    this.chooseMap = this.chooseMap.bind(this);
  }

  async componentDidMount() {
    let response = await axios.get('http://localhost:3000/data/locations.json');
    this.setState({
      locations: response.data.locations
    })
  }

  chooseMap(e) {
    this.setState({ currentMap: mapChooser(e.target.value) });
  }

  render() {
    let locationsLived = this.state.locations.map((location, id) => {
      return (<Button handleClick={this.chooseMap} key={id} city={location.city} />)
    });

    return (
      <div>
        <Header />
        <div>
          {locationsLived}
        </div>
        <Map imagename={this.state.currentMap} location={this.props.location} />
      </div>
    );
  }
}

export default MyLocations;