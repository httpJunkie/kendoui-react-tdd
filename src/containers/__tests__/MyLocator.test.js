import React from 'react';
import { shallow } from 'enzyme';
import MyLocations from '../MyLocations';
import axios from 'axios';
import renderer from 'react-test-renderer';

describe("MyLocations", function () {

  let mountedMyLocations;
  beforeEach(() => {
    mountedMyLocations = shallow(<MyLocations />);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<MyLocations />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls axios.get in #componentDidMount', () => {
    return mountedMyLocations.instance().componentDidMount().then(() => {
      expect(axios.get).toHaveBeenCalled();
    })
  });

  it('calls axios.get with correct url', () => {
    return mountedMyLocations.instance().componentDidMount().then(() => {
      expect(axios.get).toHaveBeenCalledWith('http://localhost:3000/data/locations.json');
    })
  });

  it('updates state with api data', () => {
    return mountedMyCities.instance().componentDidMount().then(() => {
      expect(mountedMyCities.state()).toHaveProperty('locations',
        [{
          "city": "test city",
          "address": "test address"
        }]
      );
    })
  });

  it('renders without crashing', () => {
    let mountedMyLocations = shallow(<MyLocations />);
  });

  it('renders a header', () => {
    const headers = mountedMyLocations.find('Header');
    expect(headers.length).toBe(1);
  });


  it('renders a map', () => {
    const maps = mountedMyLocations.find('Map');
    expect(maps.length).toBe(1);
  })
});

describe('chooseMap', () => {
  it('updates this.state.currentMap using the city passed to it', () => {
    let mountedMyLocations = shallow(<MyLocations />);
    let mockEvent = { target: { value: 'testland' } };
    mountedMyLocations.instance().chooseMap(mockEvent);
    expect(mountedMyLocations.instance().state.currentMap).toBe('testland.png');
  })
});