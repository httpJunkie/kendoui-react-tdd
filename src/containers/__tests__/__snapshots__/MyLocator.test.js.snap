// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyLocations renders correctly 1`] = `
<div>
  <div
    className="Header"
  >
    <img
      alt="Eric Bishard"
      src="images/ebishard.png"
    />
  </div>
  <div />
  <div
    className="MapBox"
  >
    <img
      alt={undefined}
      src="images/none.png"
    />
  </div>
</div>
`;
